docs: ✏️ added docs for fleek functions via sdk and fixed troubleshooting guide (#492)
fleek-platform/website: src/content/docs/SDK/Fleek_Functions/index.md

---
order: 5
title: Fleek Functions
date: 2025-02-06
desc: Create, deploy, and manage serverless functions using the Fleek SDK.
keywords: [services, Fleek SDK, serverless functions]
tags:
  - Guide
  - Learn
  - Fleek
---

# Managing Fleek Functions with the Fleek SDK

> The Fleek SDK allows you to create, deploy, and manage Fleek Functions. The SDK enables you to within your applications be able to create functions, and deploy them to the Fleek platform. The following documentation provides the methods available to interface with Fleek Functions service.

Reviewer: Sounds redundant. You start by saying what the user's able to do, to then say they are able to create functions and deploy. I think you can keep it simple, something like:

```suggestion
The Fleek SDK lets you create, deploy, and manage Fleek Functions within your applications. This document outlines the available methods for interacting with the Fleek Functions service.
```

:::note
You need to [authenticate to Fleek](/docs/sdk/#authentication) via the Fleek SDK.
:::

The `FunctionsClient` class provides an interface for managing Fleek Functions, enabling users to create, deploy, update, list, and delete serverless functions via the Fleek SDK.

## Methods

### `get`

Retrieve a specific Fleek Function by its name.

#### **Function Signature**

```typescript
get({ name }: GetFleekFunctionArgs): Promise<FleekFunction>
```

#### **Parameters**

- `name` (string) - The name of the Fleek Function to retrieve.

#### **Returns**

A `Promise` resolving to an object containing:

- Function metadata, including `id`, `name`, `slug`, `invokeUrl`, `projectId`, `status`, and `createdAt`.
- Deployment details, including `currentDeployment` and `deployments`.

> #### **Usage Example**

Reviewer: Remember the comment about the ** in the titles? You can see it here...

```suggestion
#### Usage Example
```

---
order: 5
title: Fleek Functions
date: 2025-02-06
desc: Create, deploy, and manage serverless functions using the Fleek SDK.
keywords: [services, Fleek SDK, serverless functions]
tags:

>   - Accounts

Reviewer: ```suggestion
```

  - Guide
  - Learn
  - Fleek
---

# Managing Fleek Functions with the Fleek SDK

> The Fleek SDK allows you to create, deploy, and manage Fleek Functions. This guide will walk you through the process of setting up a project, creating functions, and deploying them to the Fleek platform.

Reviewer: The document is not a guide. There's no "walk you through"...

```suggestion
The Fleek SDK allows you to create, deploy, and manage Fleek Functions. The following documentation provides the methods available to interface with Fleek Functions service.
```
